gabbi/case.py

```diff
@@ -544,7 +544,11 @@ def assert_in_or_print_output(self, expected, iterable):
                 dumper_class = self.get_content_handler(self.content_type)
                 if dumper_class:
                     full_response = dumper_class.dumps(self.response_data,
-                                                       pretty=True)
+                                                       pretty=True, test=self)
+                    # Sometimes, a given content handler may not hand back a
+                    # string for transmission, so we should be careful.
+                    if not isinstance(full_response, six.string_types):
+                        full_response = str(full_response)
```

Comment:
Hmmm, yeah, I'm not sure about this.

Since we're working on `response_data` here, not the `data`, and using the response content-type, not the request, the semantics of `dumps()` is a bit under-defined.

Since this change isn't specifically related to the addition of the test parameter, I'd leave this out of this pull request and perhaps register a new issue for the problem that happens here, along with a failure scenario.

It is probably something worth addressing, but in this change is a bit opportunistic rather than germane.